erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -35,6 +35,9 @@ def __init__(self, *args, **kwargs):
 	def autoname(self):
 		self.name = make_autoname(self.series)
 
+	def before_save(self):
+		self.compute_year_to_date()
```

Comment:
Move this call inside validate trigger after `calculate_net_pay` method. No need to add before_save method.

Reply:
Done.